modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/ValidationManager.java

```diff
@@ -112,13 +113,28 @@ protected ValidationResult validateNode(
                     "Cluster tags do not match. Cluster tag: %s, cluster tag stored in CMG: %s",
                     clusterTag, state.clusterTag()
             ));
+        } else if (!isColocationEnabledMatched(isColocationEnabled(node))) {
+            return ValidationResult.errorResult(String.format(
+                    "Colocation enabled mode does not match. Joining node colocation mode is: %s, cluster colocation mode is: %s",
+                    isColocationEnabled(node),
+                    isColocationEnabled(logicalTopology.getLogicalTopology().nodes().iterator().next())
```

Comment:
Each `logicalTopology.getLogicalTopology()` is a storage read, please think about reducing the amount of these calls

Reply:
Having one extra storage read in case of invalid enabledColocation flag is not a problem since join is a rare operation.